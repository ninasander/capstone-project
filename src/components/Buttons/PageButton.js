import React from 'react'
import styled from 'styled-components/macro'

export default function PageButton({ buttonText, onClick }) {
  return <ButtonStyled onClick={onClick}>{buttonText}</ButtonStyled>
}

const ButtonStyled = styled.button`
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  display: block;
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);
`
