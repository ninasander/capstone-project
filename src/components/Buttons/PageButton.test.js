import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import { render } from '@testing-library/react'
import PageButton from './PageButton'

describe('PageButton', () => {
  it('renders Add Enemy button correctly', () => {
    const tree = renderer.create(<PageButton buttonText="Add Enemy" />)
    expect(tree).toMatchSnapshot()
  })
  it('renders Add Player button correctly', () => {
    const tree = renderer.create(<PageButton buttonText="Add Player" />)
    expect(tree).toMatchSnapshot()
  })
  it('renders Start Encounter button correctly', () => {
    const tree = renderer.create(<PageButton buttonText="Start Encounter" />)
    expect(tree).toMatchSnapshot()
  })
  it('renders End Encounter button correctly', () => {
    const tree = renderer.create(<PageButton buttonText="End Encounter" />)
    expect(tree).toMatchSnapshot()
  })
  it('calls onClick when button is clicked', () => {
    const onClickMock = jest.fn()

    const { getByText } = render(
      <PageButton onClick={onClickMock} buttonText="Add Enemy" />
    )
    const button = getByText('Add Enemy')
    userEvent.click(button)

    expect(onClickMock).toHaveBeenCalled()
  })
})
