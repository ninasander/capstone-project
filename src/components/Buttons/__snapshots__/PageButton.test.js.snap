// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButton renders Add Enemy button correctly 1`] = `
.c0 {
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  display: block;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

<button
  className="c0"
>
  Add Enemy
</button>
`;

exports[`PageButton renders Add Player button correctly 1`] = `
.c0 {
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  display: block;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

<button
  className="c0"
>
  Add Player
</button>
`;

exports[`PageButton renders End Encounter button correctly 1`] = `
.c0 {
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  display: block;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

<button
  className="c0"
>
  End Encounter
</button>
`;

exports[`PageButton renders Start Encounter button correctly 1`] = `
.c0 {
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  display: block;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

<button
  className="c0"
>
  Start Encounter
</button>
`;
